contracts/JBNFTRewardDataSourceDelegate.sol

```diff
@@ -0,0 +1,401 @@
+  */
+  uint256 private _nextTokenId;
+
+  /**
+    @notice
+    Current supply.
+  */
+  uint256 private _supply;
+
+  /**
+    @notice
+    Token base uri.
+  */
+  string private _baseUri;
+
+  /**
+    @notice
+    Custom token uri resolver, superceeds base uri.
+  */
+  IJBToken721UriResolver private _tokenUriResolver;
+
+  /**
+    @notice
+    Contract Opensea-style metadata uri.  Learn more https://bit.ly/3NnBa9v.
+  */
+  string private _contractUri;
+
+  /**
+    @param projectId JBX project id this reward is associated with.
+    @param directory JBX directory.
+    @param maxSupply Total number of reward tokens to distribute.
+    @param minContribution Minimum contribution amount to be eligible for this reward.
+    @param _name The name of the token.
+    @param _symbol The symbol that the token should be represented by.
+    @param _uri Token base URI.
+    @param _tokenUriResolverAddress Custom uri resolver.
+    @param _contractMetadataUri Opensea-style contract metadata uri. 
+    @param _admin Set an alternate owner.
+  */
+  constructor(
+    uint256 projectId,
+    IJBDirectory directory,
+    uint256 maxSupply,
+    JBTokenAmount memory minContribution,
+    string memory _name,
+    string memory _symbol,
+    string memory _uri,
+    IJBToken721UriResolver _tokenUriResolverAddress,
+    string memory _contractMetadataUri,
+    address _admin
+  ) ERC721Rari(_name, _symbol) {
+    // JBX
+    _projectId = projectId;
+    _directory = directory;
+    _maxSupply = maxSupply;
+    _minContribution = minContribution;
+
+    // ERC721
+    _baseUri = _uri;
+    _tokenUriResolver = _tokenUriResolverAddress;
+    _contractUri = _contractMetadataUri;
+
+    if (_admin != address(0)) _transferOwnership(_admin);
+  }
+
+  //*********************************************************************//
+  // ------------------- IJBFundingCycleDataSource --------------------- //
+  //*********************************************************************//
+
+  function payParams(JBPayParamsData calldata _data) 
+    external 
+    view 
+    override 
+    returns (uint256 weight, string memory memo, IJBPayDelegate delegate) 
+  {
+    return (0, _data.memo, IJBPayDelegate(address(this)));
+  }
+
+  function redeemParams(JBRedeemParamsData calldata _data)
+    external
+    pure
+    override
+    returns (uint256 reclaimAmount, string memory memo, IJBRedemptionDelegate delegate)
+  {
+    return (0, _data.memo, IJBRedemptionDelegate(address(0)));
+  }
+
+  //*********************************************************************//
+  // ------------------------ IJBPayDelegate --------------------------- //
+  //*********************************************************************//
+
+  function didPay(JBDidPayData calldata _data) 
+    external 
+    override 
+  {
+    if (!_directory.isTerminalOf(_projectId, IJBPaymentTerminal(msg.sender))) revert INVALID_PAYMENT_EVENT();
+
+    if (_distributedSupply == _maxSupply) return;    
```

Comment:
with the weight of 0 supra and this check, contributors would then pay without anything in return (no more NFT and no projct token to at least claim back their eth) -> per design?